library/std/src/sys/pal/unix/thread.rs

```diff
@@ -847,10 +847,37 @@ pub mod guard {
             let stackptr = get_stack_start_aligned()?;
             let guardaddr = stackptr.addr();
             // Technically the number of guard pages is tunable and controlled
-            // by the security.bsd.stack_guard_page sysctl, but there are
-            // few reasons to change it from the default. The default value has
-            // been 1 ever since FreeBSD 11.1 and 10.4.
-            const GUARD_PAGES: usize = 1;
+            // by the security.bsd.stack_guard_page sysctl.
+            // By default it is 1, checking once is enough since it is
+            // a boot time config value.
+            static ONCE: crate::sync::Once = crate::sync::Once::new();
+            static mut GUARD_PAGES: usize = 1;
```

Comment:
This is just a [`OnceLock`](https://doc.rust-lang.org/nightly/std/sync/struct.OnceLock.html), right?